artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/PostOfficeImpl.java

```diff
@@ -571,6 +571,8 @@ public AddressInfo getAddressInfo(SimpleString addressName) {
    // even though failover is complete
    @Override
    public synchronized void addBinding(final Binding binding) throws Exception {
+      server.callBrokerPlugins(server.hasBrokerPlugins() ? plugin -> plugin.beforeAddBinding(binding) : null);
```

Comment:
Due to the `default` uses on `callBrokerPlugins` and other wierd reasons dependent by how the JIT works with compilations/inlining/interface optimization would be better at least to turn it into:
```
if(server.hasBrokerPlugins()){
 ....
}
```
I've noticed it is used on most "cold path" of the broker so it is fine to left as it is in this place too: it doesn't seem a hot path. My suggestion is just for wider puporses :+1:

Reply:
I can change it everywhere the plugins are called.  There are some hot paths where the plugins are called such as during message routing so if that change will improve performance then that is fine.